response_operations_ui/views/accounts.py

```diff
@@ -104,66 +101,97 @@ def change_account_name():
+                    "A verification email has been sent.",
                     category="successful_signout",
                 )
                 return redirect(url_for("logout_bp.logout"))
-            except NotifyError as e:
-                logger.error("Error sending 'email change' email to Notify Gateway", msg=e.description)
-                flash("Something went wrong while updating your email. Please try again", category="error")
+
+            flash("Something went wrong while updating your email. Please try again", category="error")
         else:
-            return redirect(url_for("account_bp.get_my_account"))
+            logger.info(f"Unable to change username and email for {user_id}, email already used")
+            flash("Email already in use", category="error")
+
     return render_template("account/change-email.html", form=form, errors=form.errors)
 
 
+def _notify_account_update_emails_sent(current_email: str, new_email: str, first_name: str, user_id: str) -> bool:
+    notify_controller = NotifyController()
+
+    update_account_details_personalisation = {
+        "first_name": first_name,
+        "value_name": "email",
+        "changed_value": new_email,
+    }
+    token = token_decoder.generate_token(json.dumps({"email": new_email, "user_id": user_id}))
+    verification_url = f"{app.config['RESPONSE_OPERATIONS_UI_URL']}/account/verify-email/{token}"
+    update_email_personalisation = {"CONFIRM_EMAIL_URL": verification_url, "first_name": first_name}
+    logger.info("Sending update account verification email", verification_url=verification_url)
+
+    try:
+        notify_controller.request_to_notify(
+            email=new_email, template_name="update_email", personalisation=update_email_personalisation
+        )
+        notify_controller.request_to_notify(
+            email=current_email,
+            template_name="update_account_details",
+            personalisation=update_account_details_personalisation,
+        )
+    except NotifyError as e:
+        logger.error(
+            f"Error sending change of email acknowledgement to Notify Gateway for {user_id}", msg=e.description
+        )
+        return False
+
+    return True
+
+
+def _can_change_username_and_email(new_email: str) -> bool:
+    user_filter = f"email+eq+%22{new_email}%22+or+userName+eq+%22{new_email}%22"
+    uaa_user = uaa_controller.get_user_by_filter(user_filter)
+
+    if uaa_user and uaa_user["totalResults"] == 0:
+        return True
+
+    return False
+
+
 @account_bp.route("/verify-email/<token>", methods=["GET"])
-@login_required
 def verify_email(token):
     try:
         duration = app.config["UPDATE_ACCOUNT_EMAIL_TOKEN_EXPIRY"]
         json_token = token_decoder.decode_email_token(token, duration)
         token_dict = json.loads(json_token)
-        user_id = session["user_id"]
-        if token_dict["user_id"] == user_id:
```

Comment:
This has been checked with IA, we don't need them to login on confirming an email/username change